src/tribler-core/tribler_core/components/tag/restapi/tags_endpoint.py

```diff
@@ -28,27 +29,35 @@ class TagsEndpoint(RESTEndpoint):
     Top-level endpoint for tags.
     """
 
-    def __init__(self, db: TagDatabase, community: TagCommunity):
+    def __init__(self, db: TagDatabase, community: TagCommunity, mds: MetadataStore):
         super().__init__()
         self.db: TagDatabase = db
         self.community: TagCommunity = community
+        self.mds: MetadataStore = mds
 
     @staticmethod
-    def validate_infohash(infohash: str) -> Tuple[bool, Optional[RESTResponse]]:
+    def validate_infohash(infohash: str) -> Optional[RESTResponse]:
         try:
-            infohash = unhexlify(infohash)
-            if len(infohash) != 20:
-                return False, RESTResponse({"error": "Invalid infohash"}, status=HTTP_BAD_REQUEST)
+            if len(unhexlify(infohash)) != 20:
+                return RESTResponse({"error": "Invalid infohash"}, status=HTTP_BAD_REQUEST)
         except binascii.Error:
-            return False, RESTResponse({"error": "Invalid infohash"}, status=HTTP_BAD_REQUEST)
+            return RESTResponse({"error": "Invalid infohash"}, status=HTTP_BAD_REQUEST)
+        return None
 
-        return True, None
+    @staticmethod
+    def validate_tag(tag: str) -> Optional[RESTResponse]:
+        try:
+            validate_tag(tag)
+        except AssertionError as e:
+            return RESTResponse({"error": f'{e}'}, status=HTTP_BAD_REQUEST)
+        return None
 
     def setup_routes(self):
         self.app.add_routes(
             [
                 web.patch('/{infohash}', self.update_tags_entries),
                 web.get('/{infohash}/suggestions', self.get_suggestions),
+                web.get('/search', self.search),
```

Comment:
I think we should have two different endpoints because in the case of FTS search we use the MDS database. In the case of search by tags, we use TagsDatabase.

As well as databases, logic in both cases is different. 
So, for me, it looks like we definitely should have two separate endpoints instead of having just one.

Reply:
The separation between MDS and Tags database is, in my opinion, just an implementation detail of Core, and later we can join them into a single database. It is not necessary to manifest this separation in REST API. We already have endpoints that are connected to multiple components. For example, `DownloadsEndpoint` is connected to the download manager, MDS, and tunnel community. `ChannelsEndpoint` is connected to MDS and Tags databases, as well as resources provided by three other components. So, there is nothing special if `SearchEndpoint` will connect to both MDS and Tags databases and combine data from both databases to serve a single API request.

Reply:
Fixed